src/main/java/io/cdap/plugin/http/common/pagination/BaseHttpPaginationIterator.java

```diff
@@ -122,8 +130,13 @@ protected BasePage getNextPage() throws IOException {
       case SUCCESS:
         break;
       case STOP:
-        throw new IllegalStateException(String.format("Fetching from url '%s' returned status code '%d' and body '%s'",
-                                                      nextPageUrl, httpStatusCode, response.getBody()));
+        String errorMessage = String.format("Fetching from url '%s' returned status code '%d' and body '%s'",
+                nextPageUrl, httpStatusCode, response.getBody());
+        throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+                errorMessage, String.format("Error message: %s", errorMessage),
+                ErrorUtils.getActionErrorByStatusCode(httpStatusCode).getErrorType(),
+                true, ErrorCodeType.HTTP, String.valueOf(httpStatusCode),
+                "https://developer.mozilla.org/en-US/docs/Web/HTTP/Status", new IllegalStateException(errorMessage));
```

Comment:
we should use official documentations link: https://datatracker.ietf.org/doc/html/rfc7231#section-6

Reply:
updated